website/addons/gdrive/utils.py

```diff
@@ -0,0 +1,169 @@
+            self.node.save()
+
+
+def serialize_urls(node_settings):
+    node = node_settings.owner
+    return {
+        'create': node.api_url_for('drive_oauth_start'),
+        'importAuth': node.api_url_for('gdrive_import_user_auth'),
+        'deauthorize': node.api_url_for('gdrive_deauthorize'),
+        'get_folders': node.api_url_for('gdrive_folders', includeRoot=1),
+        'config': node.api_url_for('gdrive_config_put'),
+        'files': node.web_url_for('collect_file_trees'),
+    }
+
+
+def serialize_settings(node_settings, current_user):
+    """
+    View helper that returns a dictionary representation of a GdriveNodeSettings record. Provides the return value for the gdrive config endpoints.
+    """
+    user_settings = node_settings.user_settings
+    user_is_owner = user_settings is not None and (
+        user_settings.owner._primary_key == current_user._primary_key
+    )
+    current_user_settings = current_user.get_addon('gdrive')
+    ret = {
+        'nodeHasAuth': node_settings.has_auth,
+        'userIsOwner': user_is_owner,
+        'userHasAuth': current_user_settings is not None and current_user_settings.has_auth,
+        'urls': serialize_urls(node_settings)
+    }
+    if node_settings.has_auth:
+        # Add owner's profile URL
+        ret['urls']['owner'] = web_url_for('profile_view_id', uid=user_settings.owner._id)
+        ret['ownerName'] = user_settings.owner.fullname
+        ret['access_token'] = user_settings.access_token
+        ret['currentFolder'] = node_settings.folder  # if node_settings.folder else None
+    return ret
+
+
+def clean_path(path):
+    """Ensure a path is formatted correctly for url_for."""
+    if path is None:
+        return ''
+    parts = path.strip('/').split('/')
+    if len(parts) <= 1:
+        cleaned_path = '/' + parts[len(parts) - 1]
+    cleaned_path = parts[len(parts) - 1]
+
+    return cleaned_path
+
+
+def build_gdrive_urls(item, node, path):
+    return {
+        'get_folders': node.api_url_for('gdrive_folders', folderId=item['id'], path=path),
+        'fetch': node.api_url_for('gdrive_folders', folderId=item['id']),
+    }
+
+
+def to_hgrid(item, node, path):
+    """
+    :param item: contents returned from Google Drive API
+    :return: results formatted as required for Hgrid display
+    """
+    path = os.path.join(path, item['title'])
+    serialized = {
+        'addon': 'gdrive',
+        'name': item['title'],
+        'id': item['id'],
+        'kind': 'folder',
+        'urls': build_gdrive_urls(item, node, path=path),
+        'path': path,
+    }
+    return serialized
+
+
+def check_access_token(user_settings):
+    cur_time_in_millis = time.mktime(datetime.datetime.utcnow().timetuple())
+    if user_settings.token_expiry <= cur_time_in_millis:
+        refresh_access_token(user_settings)
+        # return{'status': 'token_expired'}
+    return {'status': 'token_valid'}
+
+
+def refresh_access_token(user_settings):
+    try:
+        params = {
```

Comment:
This `try` block should be tightened up. It should only include the code which you expect to raise an error.

Also, as noted in the `TODO`, you should catch a specific `Exception` type.